### ISSUE TITLE
Opt-out Persistence Broken in 3.0.0.beta

### ISSUE THREAD

**Issue by nibe** [mixpanel/mixpanel-swift#457]
`hasOptedOutTracking()` always returns `false` on launch and `track` sends events even if the user opted out before.

Example code:
```swift
func applicationDidFinishLaunching(_ aNotification: Notification) {
    Mixpanel.initialize(token: YOUR_TOKEN_HERE)
    
    print("hasOptedOutTracking on applicationDidFinishLaunching: \(Mixpanel.mainInstance().hasOptedOutTracking())")
    
    if Mixpanel.mainInstance().hasOptedOutTracking() {
        Mixpanel.mainInstance().optInTracking()
    } else {
        Mixpanel.mainInstance().optOutTracking()
    }
    
    print("hasOptedOutTracking after toggling: \(Mixpanel.mainInstance().hasOptedOutTracking())")
}
```

Branch 3.0.0.beta
Compiled for macOS
macOS 11.4
Xcode 12.4

**Comment by zihejia** [comment]
hi @nibe , thanks for bringing up it, we will have a look.

**Comment by zihejia** [comment]
hi @nibe , similar to https://github.com/mixpanel/mixpanel-swift/issues/458. Now the issue has been fixed in v2.10.2. Closing it now.

**Comment by zihejia** [comment]
Updated in 3.0.0.beta branch
